jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/NativeLibLoader.java

```diff
@@ -61,13 +74,39 @@ static OS detectOS() {
       } else if (os.contains("win")) {
         return WINDOWS;
       } else if (os.contains("nux")) {
-        return LINUX;
+        return isMuslBased() ? LINUX_MUSL : LINUX;
       } else if (os.contains("sunos")) {
         return SOLARIS;
       } else {
         throw new IllegalStateException("Unsupported OS:" + os);
       }
     }
+
+    /**
+     * Checks if the Linux OS is musl based. For this, we check the memory-mapped
+     * files and see if one of those contains the string "musl".
```

Comment:
Could you update this so it says it checks the filenames, rather than the file contents. It's a little ambiguous.

Reply:
Fair point.

CLARIFIED (c3cc638f)